### PR TITLE
[RLC] support multiple mediatypes

### PR DESCRIPTION
**Pull request by joheredi**
Add support and test coverage for multiple media types

### CODE REVIEW

**Comment by qiaozha** [comment]
Just wondering if it's possible that the response could also be media types? Should we support this kind of scenarios?

**Comment by joheredi** [comment]
> Just wondering if it's possible that the response could also be media types? Should we support this kind of scenarios?

This should be handled already if the Swagger defines the options a "Content-Type" header can have. For example a swagger like this

```json
"responses": {
          "200": {
            "description": "Empty Response",
            "headers": {
              "Content-Type": {
                "description": "response with header value \"Content-Type\": \"text/html\"",
                "type": "string",
                "enum": [ "text/html" ]
              }
            }
          },
          "default": {
            "description": "Unexpected error",
            "schema": {
              "$ref": "#/definitions/Error"
            }
          }
        }
```

would end up generating a response type like this

```typescript
export interface ResponseProtectedKey200Headers {
  /** response with header value "Content-Type": "text/html" */
  "content-type"?: "text/html";
}
```

File: test/integration/generated/mediaTypesRest/src/parameters.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   * Input parameter.
+   *
+   * Value may contain any sequence of octets
+   */
+  body?: string;
+}
+
+export interface AnalyzeBodyMediaTypesParam {
+  /** Request content type */
+  contentType?: "application/pdf" | "image/jpeg" | "image/png" | "image/tiff";
```
Comment (qiaozha) [inline test/integration/generated/mediaTypesRest/src/parameters.ts:18]: Will we pass the value for the customer by default if this contentType is a single value but not equals to application/json ?

**Comment by qiaozha** [comment]
Are we suppose to make this before or after the stream support ?

File: test/integration/generated/mediaTypesRest/src/parameters.ts (hunk 1 above)
Comment (joheredi) [reply test/integration/generated/mediaTypesRest/src/parameters.ts:18]: Nope, we can't pass values by default as we are not generating any runtime code, just type definitions. We could explore making contentType required and only accept one contentType. But I would like to wait until we get feedback on that. In the meantime we'll keep it optional and default to json

**Comment by joheredi** [comment]
> Are we suppose to make this before or after the stream support ?

Before as there is still some design discussion and testing for stream support, meanwhile the response content will be loaded in memory.

File: test/integration/generated/mediaTypesRest/src/parameters.ts (hunk 1 above)
Comment (qiaozha) [reply test/integration/generated/mediaTypesRest/src/parameters.ts:18]: I see. Thanks

**Review by qiaozha — APPROVED** [review]
LGTM

**Pull request merged**